### DISCUSSION TITLE
GeoJsonLayer on data load error

### DISCUSSION THREAD

**Discussion by SchemeSonic** [visgl/deck.gl#7093] [Q&A] [answered]
Hello,  I've checked the issues but couldn't find any related discussion. So the problem is handling failed GeoJsonLayer data loading.

I' am using data URI for loading data and `onDataLoad` to get information about fetched data. However if there is any problem while loading the data from URI, console shows below error:

```
component-state.js:211 TypeError: Failed to fetch
    at _callee$ (fetch-file.js:35:1)
    at tryCatch (runtime.js:63:1)
    at Generator.invoke [as _invoke] (runtime.js:293:1)
    at Generator.next (runtime.js:118:1)
    at asyncGeneratorStep (asyncToGenerator.js:3:1)
    at _next (asyncToGenerator.js:25:1)
    at asyncToGenerator.js:32:1
    at new Promise (<anonymous>)
    at asyncToGenerator.js:21:1
    at _fetchFile (fetch-file.js:63:1)
```

I tried adding `onError` callback to DeckGL but no change. Also couldn't find any prop for GeoJsonLayer to handle load data error.

Is there any way to get error callback while loading the data from URI? I know I can load data by myself however using URI is waay easier for me.

Thanks!

**Comment by Pessimistress** [comment, marked as answer]
Both `layer.props.onError` and `deck.props.onError` are called if data fetching fails, at least on the latest version.

**Reply by SchemeSonic** [reply]
Thanks for fast reply @Pessimistress, I am using deckgl react version and I tried onError props both on layer and deckgl component however they are not triggered.  Below is my usage:

PS: What I want to achieve is setLoading(false) on data fetch error

```tsx
...
const layers = [
new GeoJsonLayer({
      id: "quality-index",
      data: dataUrl,
      ...
      onDataLoad: (data) => {
        setAnalysisResult(data);
        setLoading(false);
      },
      onError: (err) => {
        setLoading(false);
        console.log("Layer_ERROR", err); // not triggered
      },
      extensions: [
        new DataFilterExtension({ filterSize: 1, countItems: true })
      ],
      getFilterValue: (d) => parseInt(d.properties.region),
      filterRange: [parseInt(activeRegion), parseInt(activeRegion)]
    }
];

return (<DeckGL
        layers={layers}
        pickingRadius={5}
        onError={(err) => {
          setLoading(false);
          console.log("Deck_ERROR", err); // not triggered
        }}
        initialViewState={INITIAL_VIEW_STATE}
        controller={true}
      >
        <InteractiveMap
          mapboxApiAccessToken={MAPBOX_ACCESS_TOKEN}
          reuseMaps
          mapStyle="mapbox://styles/mapbox/dark-v10"
          preventStyleDiffing={true}
          ref={onMapRefChange}
        />
      </DeckGL>);
```

**Reply by Pessimistress** [reply]
Check your deck.gl version.

**Reply by SchemeSonic** [reply]
Thanks! Works in latest version.

**Reply by neelduttahere** [reply]
Same issue here, and I am using deck.gl v8.9.22

**Comment by sheecegardezi** [comment]
Same issue here, and I am using deck.gl v9.0.38. How did you resolve it?

**Reply by sheecegardezi** [reply]
I think you need to set `debug: true,` property in deckgl instance ?
